packages/manager/src/features/CloudPulse/Dashboard/CloudPulseDashboard.tsx

```diff
@@ -0,0 +1,209 @@
+  getInSeconds,
+  getIntervalIndex,
+} from '../Widget/components/CloudPulseIntervalSelect';
+
+import type { CloudPulseWidgetProperties } from '../Widget/CloudPulseWidget';
+import type {
+  AvailableMetrics,
+  Dashboard,
+  JWETokenPayLoad,
+  TimeDuration,
+  Widgets,
+} from '@linode/api-v4';
+
+export interface DashboardProperties {
+  dashboardId: number;
+  duration: TimeDuration;
+  manualRefreshTimeStamp?: number | undefined;
+  region?: string;
+  resources: string[];
+  savePref?: boolean;
+}
+
+const StyledErrorState = styled(Placeholder, {
+  label: 'StyledErrorState',
+})({
+  height: '100%',
+});
+
+export const CloudPulseDashboard = (props: DashboardProperties) => {
+  const getJweTokenPayload = () => {
+    return {
+      resource_id: resources?.map((resource) => String(resource.id)) ?? [],
+    } as JWETokenPayLoad;
+  };
+
+  const getCloudPulseGraphProperties = (widget: Widgets) => {
+    const graphProp: CloudPulseWidgetProperties = {} as CloudPulseWidgetProperties;
+    graphProp.widget = { ...widget };
+    if (props.savePref) {
+      setPrefferedWidgetPlan(graphProp.widget);
+    }
+    graphProp.serviceType = dashboard?.service_type ?? '';
+    graphProp.resourceIds = props.resources;
+    graphProp.duration = props.duration;
+    graphProp.unit = widget.unit ? widget.unit : '%';
+    graphProp.ariaLabel = widget.label;
+    graphProp.errorLabel = 'Error While loading data';
+    graphProp.timeStamp = props.manualRefreshTimeStamp!;
+
+    return graphProp;
+  };
+
+  const setPrefferedWidgetPlan = (widgetObj: Widgets) => {
+    const widgetPreferences = getUserPreferenceObject().widgets;
+    if (widgetPreferences && widgetPreferences[widgetObj.label]) {
+      const pref = widgetPreferences[widgetObj.label];
+      widgetObj.size = pref.size;
+      widgetObj.aggregate_function = pref.aggregateFunction;
+      widgetObj.time_granularity = { ...pref.timeGranularity };
+    }
+  };
+
+  const getTimeGranularity = (scrapeInterval: string) => {
+    const scrapeIntervalValue = getInSeconds(scrapeInterval);
+    const index = getIntervalIndex(scrapeIntervalValue);
+    return index < 0 ? all_interval_options[0] : all_interval_options[index];
+  };
+
+  const { data: dashboard } = useCloudPulseDashboardByIdQuery(
+    props.dashboardId!,
+    props.savePref
+  );
+
+  const { data: resources } = useResourcesQuery(
+    dashboard && dashboard.service_type ? true : false,
+    dashboard ? dashboard.service_type : undefined!,
+    {},
+    {}
+  );
+
+  const {
+    data: metricDefinitions,
+    isLoading,
+  } = useGetCloudPulseMetricDefinitionsByServiceType(
+    dashboard ? dashboard!.service_type : undefined!,
+    dashboard && dashboard!.service_type !== undefined ? true : false
+  );
+
+  const {
+    data: jweToken,
+    isError: isJweTokenError,
+  } = useCloudPulseJWEtokenQuery(
+    dashboard ? dashboard.service_type! : undefined!,
+    getJweTokenPayload(),
+    resources && dashboard ? true : false
```

Comment:
```suggestion
    Boolean(resources)
```
Resources is dependent on dashboard so that's all we really need to check I believe.

Reply:
updated